src/server/pfs/cmds/cmds.go

```diff
@@ -391,46 +433,44 @@ Put the data from a URL as repo/commit/url_path:
 					return err
 				}
 				defer func() {
-					if err := client.FinishCommit(commit.Repo.Name, commit.ID); err != nil && retErr == nil {
-						retErr = err
+					if retErr != nil {
+						// something errored so we try to cancel the commit, if
+						// we error doing this we don't report because retErr
+						// occurred first
+						_ = client.CancelCommit(commit.Repo.Name, commit.ID)
```

Comment:
While we won't return it, we may want to know if we failed to recover. If the user knows its in an undefined state, we can instruct them to at least check the state.